geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManager.java

```diff
@@ -85,11 +98,17 @@ Connection borrowConnection(ServerLocation server, long aquireTimeout, boolean o
   void close(boolean keepAlive);
 
   /**
-   * Exchange one connection for a new connection to a different server.
+   * Exchange one connection for a new connection to a different server. This method can break the
+   * max connection contract if there is no available connection and maxConnections has already been
+   * reached.
    *
    * @param conn connection to exchange. It will be returned to the pool (or destroyed if it has
    *        been invalidated).
    * @param excludedServers servers to exclude when looking for a new connection
+   * @param aquireTimeout The amount of time to wait for a connection to be available
```

Comment:
👍
Since this is a strictly internal interface it should have no downstream impact. In the short term it bring the API signature and documentation inline with the implementation and the expected behavior from the caller standpoint. I would like to see in the future if we want timeout that we normalize it across all methods and use it in an intuitive way. Currently not the case and the effort to do so may not be worth it.

Reply:
I will write a ticket about this. It does not need to be addressed as part of this PR

Reply:
https://issues.apache.org/jira/browse/GEODE-6535